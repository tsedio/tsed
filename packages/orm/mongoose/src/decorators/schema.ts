import {decoratorTypeOf, StoreMerge, useDecorators} from "@tsed/core";
import {registerProvider} from "@tsed/di";
import {Property} from "@tsed/schema";
import {SchemaTypeOptions} from "mongoose";
import {MONGOOSE_SCHEMA} from "../constants";
import {MongooseSchemaOptions} from "../interfaces";
import {getSchema, getSchemaToken} from "../utils/createSchema";

/**
 * Define a class as a Mongoose Schema ready to be used to compose other schemes and models.
 *
 * ### Example
 *
 * ```typescript
 * @MongooseSchema()
 * export class EventSchema {
 *   @Property()
 *   field: string;
 * }
 * ```
 *
 * ### Options
 *
 * - `schemaOptions` (mongoose.SchemaOptions): Option to configure the schema behavior.
 *
 * @param {MongooseSchemaOptions | undefined} options
 * @returns {(target: any) => void}
 * @decorator
 * @mongoose
 * @property
 * @class
 */
export function Schema(options?: MongooseSchemaOptions): (target: any) => void;
export function Schema(definition: SchemaTypeOptions<any>): Function;
export function Schema(options: MongooseSchemaOptions | SchemaTypeOptions<any> = {}) {
  return (...parameters: any[]) => {
    switch (decoratorTypeOf(parameters)) {
      case "property":
        return useDecorators(Property(), StoreMerge(MONGOOSE_SCHEMA, options))(parameters[0], parameters[1], parameters[2]);

      case "class":
        const {token} = getSchemaToken(parameters[0], options);

        registerProvider({
          provide: token,
          deps: [],
          useFactory() {
            return getSchema(parameters[0], options as any);
          }
        });
        break;
    }
  };
}

/**
 * Define a class as a Mongoose Schema ready to be used to compose other schemes and models.
 *
 * ### Example
 *
 * ```typescript
 * @MongooseSchema()
 * export class EventSchema {
 *   @Property()
 *   field: string;
 * }
 * ```
 *
 * ### Options
 *
 * - `schemaOptions` (mongoose.SchemaOptions): Option to configure the schema behavior.
 *
 * @param {MongooseSchemaOptions | undefined} options
 * @returns {(target: any) => void}
 * @decorator
 * @mongoose
 * @property
 * @class
 */
export function MongooseSchema(options?: MongooseSchemaOptions): (target: any) => void;
export function MongooseSchema(definition: SchemaTypeOptions<any>): Function;
export function MongooseSchema(options: MongooseSchemaOptions | SchemaTypeOptions<any> = {}) {
  return Schema(options as any);
}
