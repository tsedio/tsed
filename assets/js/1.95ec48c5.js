(window.webpackJsonp=window.webpackJsonp||[]).push([[1,5,6],{740:function(t,e,s){},741:function(t,e,s){},742:function(t,e,s){"use strict";s(740)},743:function(t,e,s){"use strict";s.r(e);s(26),s(12),s(27);var a=s(96),i=s(11),o={name:"Page",components:{PageNav:a.PageNav,PageHeader:a.PageHeader,PageSidebar:a.PageSidebar},props:["sidebarItems"],data:function(){return{headerFixed:!1}},computed:{repoUrl:function(){return this.editUrl.replace("/edit","/blob")},hashTags:function(){return this.$frontmatter.hashTags||this.$site.themeConfig.defaultHashTags||[]},createIssueUrl:function(){var t=this.$site.themeConfig,e=t.repo,s=t.docsRepo,a=void 0===s?e:s,o=Object(i.k)(this.$page.path),r=i.l.test(a)?a:"https://github.com/".concat(a);return"".concat(r,"/issues/new?title=").concat(encodeURIComponent("[BUG] Page ".concat(this.$page.title," - ").concat(o)))},editUrl:function(){var t=this.$site.themeConfig,e=t.repo,s=t.editLinks,o=t.docsDir,r=void 0===o?"":o,n=t.docsBranch,h=void 0===n?"master":n,l=t.docsRepo,d=void 0===l?e:l,c=Object(i.k)(this.$page.path);if(i.a.test(c)?c+="README.md":c+=".md",d&&s)return Object(a.createEditLink)(e,d,r,h,c)},lastUpdated:function(){return this.$page.lastUpdated},lastUpdatedText:function(){return"string"==typeof this.$themeLocaleConfig.lastUpdated?this.$themeLocaleConfig.lastUpdated:"string"==typeof this.$site.themeConfig.lastUpdated?this.$site.themeConfig.lastUpdated:"Last Updated"},shouldShowOtherTopics:function(){var t=this.$page.frontmatter;return!t.layout&&!this.isLandingPage&&!1!==t.otherTopics&&this.otherTopicsItems.length},shouldShowPageSidebar:function(){return!1!==this.$page.frontmatter.pageSidebar},otherTopicsItems:function(){return Object(a.resolveOtherTopicsItems)(this.$page,this.$page.regularPath,this.$site,this.$localePath)}},mounted:function(){var t=this;window.addEventListener("scroll",(function(){t.headerFixed=window.pageYOffset>140}))},methods:{}},r=(s(742),s(1)),n=Object(r.a)(o,(function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("div",{ref:"page"},[s("div",[s("div",{staticClass:"relative"},[s("PageHeader",{attrs:{fixed:t.headerFixed,title:t.$page.title||"Not found","repo-url":t.repoUrl,hashTags:t.hashTags,"create-issue-url":t.createIssueUrl,"edit-url":t.editUrl}})],1),t._v(" "),s("div",{staticClass:"px-5 md:px-10 max-w-site m-auto"},[s("div",{staticClass:"page-content"},[s("article",[t._t("page-top"),t._v(" "),s("Content"),t._v(" "),t._t("page-bottom")],2),t._v(" "),s("div",{staticClass:"pt-5"},[t.lastUpdated?s("div",{staticClass:"py-5 flex justify-end text-sm"},[s("p",[s("span",{staticClass:"font-bold pr-1"},[t._v(t._s(t.lastUpdatedText)+":")]),t._v(" "),s("span",{staticClass:"time"},[t._v(t._s(t.lastUpdated))])])]):t._e(),t._v(" "),s("PageNav",{attrs:{sidebarItems:t.sidebarItems}})],1),t._v(" "),t.shouldShowOtherTopics?s("OtherTopics",{staticClass:"px-3",attrs:{slot:"page-bottom",items:t.otherTopicsItems},slot:"page-bottom"},[s("h4",{staticClass:"text-xl mb-4 mt-2 font-semibold",attrs:{slot:"top"},slot:"top"},[t._v("\n            Other topics\n          ")])]):t._e()],1)]),t._v(" "),t.shouldShowPageSidebar?s("PageSidebar",{staticClass:"fixed overflow-auto",class:{"-mini-header":t.headerFixed},attrs:{page:t.$page}}):t._e()],1)])}),[],!1,null,null,null);e.default=n.exports},744:function(t,e,s){"use strict";s(741)},747:function(t,e,s){},750:function(t,e,s){"use strict";s.r(e);s(26),s(12),s(27);var a=s(96),i=s(11),o={name:"NotFound",components:{PageNav:a.PageNav,PageHeader:a.PageHeader,PageSidebar:a.PageSidebar},props:["sidebarItems"],data:function(){return{headerFixed:!1}},computed:{repoUrl:function(){return this.editUrl.replace("/edit","/blob")},hashTags:function(){return this.$frontmatter.hashTags||this.$site.themeConfig.defaultHashTags||[]},createIssueUrl:function(){var t=this.$site.themeConfig,e=t.repo,s=t.docsRepo,a=void 0===s?e:s,o=Object(i.k)(this.$page.path),r=i.l.test(a)?a:"https://github.com/".concat(a);return"".concat(r,"/issues/new?title=").concat(encodeURIComponent("[BUG] Page ".concat(this.$page.title," - ").concat(o)))},editUrl:function(){var t=this.$site.themeConfig,e=t.repo,s=t.editLinks,o=t.docsDir,r=void 0===o?"":o,n=t.docsBranch,h=void 0===n?"master":n,l=t.docsRepo,d=void 0===l?e:l,c=Object(i.k)(this.$page.path);if(i.a.test(c)?c+="README.md":c+=".md",d&&s)return Object(a.createEditLink)(e,d,r,h,c)},lastUpdated:function(){return this.$page.lastUpdated},lastUpdatedText:function(){return"string"==typeof this.$themeLocaleConfig.lastUpdated?this.$themeLocaleConfig.lastUpdated:"string"==typeof this.$site.themeConfig.lastUpdated?this.$site.themeConfig.lastUpdated:"Last Updated"},shouldShowOtherTopics:function(){var t=this.$page.frontmatter;return!t.layout&&!this.isLandingPage&&!1!==t.otherTopics&&this.otherTopicsItems.length},otherTopicsItems:function(){return Object(a.resolveOtherTopicsItems)(this.$page,this.$page.regularPath,this.$site,this.$localePath)}},mounted:function(){var t=this;window.addEventListener("scroll",(function(){t.headerFixed=window.pageYOffset>140}))},methods:{}},r=(s(744),s(1)),n=Object(r.a)(o,(function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("div",{ref:"page"},[s("article",[s("div",{staticClass:"relative"},[s("PageHeader",{attrs:{fixed:t.headerFixed,title:"Page not found","repo-url":t.repoUrl,hashTags:t.hashTags,"create-issue-url":t.createIssueUrl,"edit-url":t.editUrl}})],1),t._v(" "),s("div",{staticClass:"px-5 md:px-10 max-w-site m-auto"},[s("div",{staticClass:"page-content"},[s("div",{staticClass:"content__default pb-10"},[s("h2",[t._v("\n            That's a Four-Oh-Four.\n          ")]),t._v(" "),s("p",[s("Button",{attrs:{href:"/"}},[t._v("\n              Go home\n            ")])],1)]),t._v(" "),s("div",{staticClass:"pt-5"},[t.lastUpdated?s("div",{staticClass:"py-5 flex justify-end text-sm"},[s("p",[s("span",{staticClass:"font-bold pr-1"},[t._v(t._s(t.lastUpdatedText)+":")]),t._v(" "),s("span",{staticClass:"time"},[t._v(t._s(t.lastUpdated))])])]):t._e(),t._v(" "),s("PageNav",{attrs:{sidebarItems:t.sidebarItems}})],1),t._v(" "),t.shouldShowOtherTopics?s("OtherTopics",{staticClass:"px-3",attrs:{slot:"page-bottom",items:t.otherTopicsItems},slot:"page-bottom"},[s("h4",{staticClass:"text-xl mb-4 mt-2 font-semibold",attrs:{slot:"top"},slot:"top"},[t._v("\n            Other topics\n          ")])]):t._e()],1)]),t._v(" "),s("PageSidebar",{staticClass:"fixed",class:{"-mini-header":t.headerFixed},attrs:{page:t.$page}})],1)])}),[],!1,null,null,null);e.default=n.exports},754:function(t,e,s){"use strict";s(747)},763:function(t,e,s){"use strict";s.r(e);var a=s(96),i=s(4),o=s(178),r=s(750);s(743);i.a.use(o.a);var n={components:{NotFound:r.default,Navbar:a.Navbar,Sidebar:a.Sidebar,Footer:a.Footer,SupportUs:a.SupportUs},data:function(){return{isSidebarOpen:!1}},computed:{navLinks:function(){return Object(a.getUserNavLinks)(this)},repoUrl:function(){var t=this.$site.themeConfig.repo;return/^https?:/.test(t)?t:"https://github.com/".concat(t)},siteTitle:function(){return this.$site.themeConfig.shortTitle||this.$siteTitle},htmlTitle:function(){return this.$site.themeConfig.htmlTitle||this.$siteTitle},logoSrc:function(){return this.$site.themeConfig.logo&&this.$withBase(this.$site.themeConfig.logo)},isLandingPage:function(){return"Home"===this.$page.frontmatter.layout},isCustomLayout:function(){return this.$page.frontmatter.layout&&!this.isLandingPage},shouldShowNavbar:function(){var t=this.$site.themeConfig;return!1!==this.$page.frontmatter.navbar&&!1!==t.navbar&&(this.$title||t.logo||t.repo||t.nav||this.$themeLocaleConfig.nav)},shouldShowSidebar:function(){return!1!==this.$page.frontmatter.sidebar&&this.sidebarItems.length},sidebarItems:function(){return Object(a.resolveSidebarItems)(this.$page,this.$page.regularPath,this.$site,this.$localePath)},pageClasses:function(){var t=this.$page.frontmatter.pageClass;return[{"no-navbar":!this.shouldShowNavbar,"sidebar-open":this.isSidebarOpen,"no-sidebar":!this.shouldShowSidebar,"with-sidebar":this.shouldShowSidebar},t]},algolia:function(){return this.$themeLocaleConfig.algolia||this.$site.themeConfig.algolia||{}}},mounted:function(){var t=this;this.$router.afterEach((function(){t.isSidebarOpen=!1}))},methods:{toggleSidebar:function(t){this.isSidebarOpen="boolean"==typeof t?t:!this.isSidebarOpen,this.$emit("toggle-sidebar",this.isSidebarOpen)},onTouchStart:function(t){this.touchStart={x:t.changedTouches[0].clientX,y:t.changedTouches[0].clientY}},onTouchEnd:function(t){var e=t.changedTouches[0].clientX-this.touchStart.x,s=t.changedTouches[0].clientY-this.touchStart.y;Math.abs(e)>Math.abs(s)&&Math.abs(e)>40&&(e>0&&this.touchStart.x<=80?this.toggleSidebar(!0):this.toggleSidebar(!1))}}},h=(s(754),s(1)),l=Object(h.a)(n,(function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("div",{staticClass:"theme-container init",class:t.pageClasses,on:{touchstart:t.onTouchStart,touchend:t.onTouchEnd}},[t.shouldShowNavbar?s("Navbar",{attrs:{"site-title":t.siteTitle,"html-title":t.htmlTitle,"logo-src":t.logoSrc,href:t.$localePath,"repo-url":t.repoUrl,"social-urls":t.$site.themeConfig,items:t.navLinks,algolia:t.algolia},scopedSlots:t._u([{key:"sidebar-before",fn:function(){return[t.shouldShowSidebar?s("div",{staticClass:"flex cursor-pointer text-xl mr-4",on:{click:t.toggleSidebar}},[s("i",{staticClass:"bx bx-menu"})]):t._e()]},proxy:!0}],null,!1,1423491846)}):t._e(),t._v(" "),s("div",{staticClass:"sidebar-mask",on:{click:function(e){return t.toggleSidebar(!1)}}}),t._v(" "),s("main",{staticClass:"main-content relative z-2 pt-16"},[t.isCustomLayout?s("div",{staticClass:"custom-layout"},[s(t.$page.frontmatter.layout,{tag:"component"})],1):s("NotFound",{attrs:{"sidebar-items":t.sidebarItems}})],1),t._v(" "),t.shouldShowSidebar?s("Sidebar",{class:{"-translate-x-100 md:shadow-sidebar md:translate-x-0":!t.isSidebarOpen,"translate-0 shadow-sidebar":t.isSidebarOpen},attrs:{items:t.sidebarItems},on:{"toggle-sidebar":t.toggleSidebar},scopedSlots:t._u([{key:"top",fn:function(){return[t._t("sidebar-top")]},proxy:!0},{key:"bottom",fn:function(){return[t._t("sidebar-bottom")]},proxy:!0}],null,!0)}):t._e(),t._v(" "),s("Footer",{attrs:{"repo-url":t.repoUrl,author:t.$site.themeConfig.author,"license-type":t.$site.themeConfig.licenseType,"copyright-dates":t.$site.themeConfig.copyrightDates,"social-urls":t.$site.themeConfig},scopedSlots:t._u([{key:"top",fn:function(){return[t.isLandingPage?t._e():s("SupportUs",{attrs:{brand:t.$site.themeConfig.shortTitle,"sponsor-url":t.$site.themeConfig.sponsorUrl,"license-type":t.$site.themeConfig.licenseType}}),t._v(" "),t._t("footer-top")]},proxy:!0},{key:"bottom",fn:function(){return[t._t("footer-bottom")]},proxy:!0}],null,!0)})],1)}),[],!1,null,null,null);e.default=l.exports}}]);