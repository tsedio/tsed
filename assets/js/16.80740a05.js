(window.webpackJsonp=window.webpackJsonp||[]).push([[16],{766:function(t,s,e){"use strict";e.r(s);var a={name:"SupportOptions"},o=e(1),n=Object(o.a)(a,(function(){var t=this,s=t.$createElement,e=t._self._c||s;return e("div",[e("ul",{staticClass:"reset-list flex justify-center align-center p-3 flex-wrap -mx-2"},[e("li",{staticClass:"p-2"},[e("Button",{attrs:{"bg-color":"button-white",color:"blue",title:"Become a backer or sponsor on OpenCollective",href:"https://opencollective.com/tsed"}},[e("BxIcon",{staticClass:"mr-1",attrs:{name:"bx-heart"}}),t._v("\n        OpenCollective\n      ")],1)],1),t._v(" "),e("li",{staticClass:"p-2"},[e("Button",{attrs:{"bg-color":"button-white",color:"blue",title:"Become a backer or sponsor on Github sponsor",href:"https://github.com/sponsors/Romakita"}},[e("BxIcon",{staticClass:"mr-1",attrs:{name:"bxl-github"}}),t._v("\n        Github sponsor\n      ")],1)],1),t._v(" "),e("li",{staticClass:"p-2"},[e("Button",{attrs:{"bg-color":"button-white",color:"blue",title:"Send a one-time donation",href:"https://paypal.me/romainlenzotti"}},[e("BxIcon",{staticClass:"mr-1",attrs:{name:"bxl-paypal"}}),t._v("\n        Paypal\n      ")],1)],1)]),t._v(" "),e("h4",{staticClass:"text-center font-bold"},[t._v("Or")]),t._v(" "),e("ul",{staticClass:"reset-list flex justify-center align-center p-3 mb-5"},[e("li",{staticClass:"p-2"},[e("Button",{staticClass:"typeform-share",attrs:{"data-mode":"popup",href:"/contact"}},[e("BxIcon",{staticClass:"mr-1",attrs:{name:"bx-message-alt-dots"}}),t._v("\n        Contact us directly\n      ")],1)],1)])])}),[],!1,null,null,null);s.default=n.exports}}]);