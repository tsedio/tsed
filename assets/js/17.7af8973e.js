(window.webpackJsonp=window.webpackJsonp||[]).push([[17],{751:function(t,e,r){"use strict";r.r(e);var n=r(96),o={name:"Home",components:{HeroBanner:n.HeroBanner,HeroBannerCta:n.HeroBannerCta,WordsSlider:n.WordsSlider,Feature:n.Feature,GithubContributors:n.GithubContributors,OpenCollectiveBackers:n.OpenCollectiveBackers,Button:n.Button,Showcase:n.Showcase},computed:{gettingStartedUrl:function(){var t=this.$page.frontmatter.gettingStartedUrl;return t},gettingStartedText:function(){var t=this.$page.frontmatter.gettingStartedText;return t},messengerUrl:function(){var t=this.$page.frontmatter.messengerUrl;return t},messengerText:function(){var t=this.$page.frontmatter.messengerText;return t},messengerIcon:function(){var t=this.$page.frontmatter.messengerIcon;return t},sponsorUrl:function(){var t=this.$page.frontmatter.sponsorUrl;return t},sponsorText:function(){var t=this.$page.frontmatter.sponsorText;return t},sponsorIcon:function(){var t=this.$page.frontmatter.sponsorIcon;return t},features:function(){var t=this.$page.frontmatter.features,e=void 0===t?[]:t;return e},contributors:function(){var t=this.$page.frontmatter.contributors;return t},liveDemoUrl:function(){var t=this.$page.frontmatter.liveDemoUrl;return t}}},s=r(1),a=Object(s.a)(o,(function(){var t=this,e=t.$createElement,r=t._self._c||e;return r("div",[r("HeroBanner",[r("template",{slot:"hero-brand"},[r("Content",{attrs:{"slot-key":"hero-brand"}})],1),t._v(" "),r("template",{slot:"hero-slogan"},[r("Content",{attrs:{"slot-key":"hero-slogan"}})],1),t._v(" "),r("template",{slot:"hero-content"},[r("Content",{attrs:{"slot-key":"hero-content"}})],1),t._v(" "),r("template",{slot:"hero-cta"},[r("HeroBannerCta",{attrs:{"getting-started-text":t.gettingStartedText,"getting-started-url":t.gettingStartedUrl,"messenger-text":t.messengerText,"messenger-icon":t.messengerIcon,"messenger-url":t.messengerUrl,"sponsor-text":t.sponsorText,"sponsor-icon":t.sponsorIcon,"sponsor-url":t.sponsorUrl}})],1)],2),t._v(" "),r("div",{staticClass:"bg-gray-lighter py-10"},[r("div",{staticClass:"container flex flex-col w-full max-w-md mx-auto px-3 pb-5 text-center"},[r("h2",{staticClass:"text-2xl mb-5 normal-case"},[r("Content",{attrs:{"slot-key":"testimonial-title"}})],1),t._v(" "),r("Content",{attrs:{"slot-key":"testimonial-content"}})],1)]),t._v(" "),t.features.length?r("div",[r("div",{staticClass:"container flex flex-col w-full max-w-site mx-auto px-5 py-5"},[r("div",[r("div",{staticClass:"flex flex-wrap align-center justify-center"},t._l(t.features,(function(e){return r("Feature",{staticClass:"w-full sm:w-1/2 md:w-1/3 mb-5",attrs:{title:e.title,icon:e.icon}},[r("div",{domProps:{innerHTML:t._s(e.details)}})])})),1),t._v(" "),r("div",{staticClass:"flex items-center justify-center pt-5 mb-5"},[r("div",[r("Button",{staticClass:"m-2",attrs:{rounded:"medium",href:t.gettingStartedUrl}},[t._v("\n              See more\n            ")]),t._v(" "),t.liveDemoUrl?r("Button",{staticClass:"m-2",attrs:{href:t.liveDemoUrl,rounded:"medium","bg-color":"white-button",color:"blue"}},[t._v("\n              Live demo\n            ")]):t._e()],1)]),t._v(" "),r("Content",{attrs:{"slot-key":"features-footer"}})],1)])]):t._e(),t._v(" "),t._t("page-top"),t._v(" "),r("Content"),t._v(" "),t.contributors?r("Showcase",t._b({class:t.contributors.classes,scopedSlots:t._u([{key:"showcase-cta",fn:function(){return[r("Button",{attrs:{"bg-color":t.contributors.cta.bgColor,color:t.contributors.cta.color,rounded:"medium",href:t.contributors.cta.url}},[t._v("\n        "+t._s(t.contributors.cta.label)+"\n      ")])]},proxy:!0}],null,!1,3949236416)},"Showcase",t.contributors,!1),[r("GithubContributors",t._b({},"GithubContributors",t.contributors.badge,!1))],1):t._e(),t._v(" "),r("SupportUsBlock"),t._v(" "),t._t("page-bottom")],2)}),[],!1,null,null,null);e.default=a.exports}}]);