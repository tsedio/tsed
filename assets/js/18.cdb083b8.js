(window.webpackJsonp=window.webpackJsonp||[]).push([[18],{753:function(t,e,s){"use strict";s.r(e);var n={name:"OffersPage",components:{Offers:s(96).Offers},methods:{getConfig:function(){var t=this.$frontmatter.config;if(t){var e=this.$site.themeConfig.offers[t];if(e)return e}return this.$frontmatter}}},i=s(1),r=Object(i.a)(n,(function(){var t=this.$createElement,e=this._self._c||t;return e("div",[e("Offers",this._b({staticClass:"bg-gray-200"},"Offers",this.getConfig(),!1)),this._v(" "),e("div",[e("Content",{attrs:{"slot-key":"header-raw-content"}})],1),this._v(" "),e("div",{staticClass:"px-5 md:px-10 max-w-site m-auto"},[e("div",{staticClass:"page-content"},[e("Content")],1)]),this._v(" "),e("div",[e("Content",{attrs:{"slot-key":"footer-raw-content"}})],1)],1)}),[],!1,null,null,null);e.default=r.exports}}]);