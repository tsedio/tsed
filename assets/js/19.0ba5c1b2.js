(window.webpackJsonp=window.webpackJsonp||[]).push([[19],{752:function(e,t,s){"use strict";s.r(t);var a={name:"WarehousePage",components:{Warehouse:s(96).Warehouse}},n=s(1),o=Object(n.a)(a,(function(){var e=this.$createElement,t=this._self._c||e;return t("Warehouse",{attrs:{url:this.$site.themeConfig.warehouseUrl||"https://api.tsed.io/rest/warehouse","initial-plugins":this.$site.themeConfig.warehouseItems}},[t("Content",{attrs:{"slot-key":"warehouse-header"}})],1)}),[],!1,null,null,null);t.default=o.exports}}]);