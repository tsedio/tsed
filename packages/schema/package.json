{
  "name": "@tsed/schema",
  "version": "6.64.2",
  "description": "JsonSchema module for Ts.ED Framework",
  "private": false,
  "source": "./src/index.ts",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "exports": {
    "require": "./lib/index.js",
    "default": "./lib/index.modern.js"
  },
  "keywords": [
    "TypeScript",
    "decorators",
    "models",
    "json schema",
    "JsonSchema",
    "class",
    "classes",
    "tsed"
  ],
  "scripts": {
    "build": "microbundle --target node --no-compress --format modern,cjs --tsconfig ./tsconfig.compile.json"
  },
  "dependencies": {
    "@tsed/core": "6.64.2",
    "@tsed/openspec": "6.64.2",
    "change-case": "4.1.2",
    "micromatch": "4.0.2",
    "tslib": "2.2.0"
  },
  "devDependencies": {
    "@apidevtools/swagger-parser": "10.0.2",
    "@types/fs-extra": "9.0.6",
    "@types/json-schema": "7.0.7",
    "@types/micromatch": "4.0.1",
    "fs-extra": "9.0.1"
  },
  "peerDependencies": {
    "@tsed/core": "6.60.1",
    "@tsed/openspec": "6.60.1"
  }
}