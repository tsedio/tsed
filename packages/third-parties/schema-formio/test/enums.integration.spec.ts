import {Enum} from "@tsed/schema";
import {DataSourceJson, getFormioSchema, Select} from "../src/index.js";

enum TestEnum {
  VALUE1 = "VALUE1",
  VALUE2 = "VALUE2"
}

describe("Enum", () => {
  it("should generate the correct schema with enum", async () => {
    class Test {
      @Enum(TestEnum)
      test: TestEnum;
    }

    const form = await getFormioSchema(Test);

    expect(form).toMatchSnapshot();
  });

  it("should generate the correct schema with select", async () => {
    class Test {
      @Enum(TestEnum)
      @Select()
      test: TestEnum;
    }

    const form = await getFormioSchema(Test);

    expect(form).toMatchSnapshot();
  });

  it("should generate the correct schema with select custom value", async () => {
    class Test {
      @Enum(TestEnum)
      @Select()
      @DataSourceJson(Object.values(TestEnum).map((value) => ({label: `label ${value}`, value})))
      test: TestEnum;
    }

    const form = await getFormioSchema(Test);

    expect(form).toMatchSnapshot();
  });
});
