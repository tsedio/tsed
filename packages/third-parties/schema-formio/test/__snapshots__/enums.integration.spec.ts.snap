// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Enum > should generate the correct schema with enum 1`] = `
{
  "access": [],
  "components": [
    {
      "disabled": false,
      "key": "test",
      "label": "Test",
      "type": "radio",
      "validate": {
        "required": false,
      },
      "values": [
        {
          "label": "VALUE1",
          "shortcut": "",
          "value": "VALUE1",
        },
        {
          "label": "VALUE2",
          "shortcut": "",
          "value": "VALUE2",
        },
      ],
    },
  ],
  "display": "form",
  "machineName": "test",
  "name": "test",
  "submissionAccess": [],
  "tags": [],
  "title": "Test",
  "type": "form",
}
`;

exports[`Enum > should generate the correct schema with select 1`] = `
{
  "access": [],
  "components": [
    {
      "data": {
        "json": "[{"label":"VALUE1","value":"VALUE1"},{"label":"VALUE2","value":"VALUE2"}]",
      },
      "dataSrc": "json",
      "disabled": false,
      "idPath": "value",
      "input": true,
      "key": "test",
      "label": "Test",
      "selectThreshold": 0.3,
      "template": "<span>{{ item.label }}</span>",
      "type": "select",
      "validate": {
        "required": false,
      },
      "valueProperty": "value",
      "widget": "choicesjs",
    },
  ],
  "display": "form",
  "machineName": "test",
  "name": "test",
  "submissionAccess": [],
  "tags": [],
  "title": "Test",
  "type": "form",
}
`;

exports[`Enum > should generate the correct schema with select custom value 1`] = `
{
  "access": [],
  "components": [
    {
      "data": {
        "json": "[{"label":"label VALUE1","value":"VALUE1"},{"label":"label VALUE2","value":"VALUE2"}]",
      },
      "dataSrc": "json",
      "disabled": false,
      "idPath": "value",
      "input": true,
      "key": "test",
      "label": "Test",
      "selectThreshold": 0.3,
      "template": "<span>{{ item.label }}</span>",
      "type": "select",
      "validate": {
        "required": false,
      },
      "valueProperty": "value",
      "widget": "choicesjs",
    },
  ],
  "display": "form",
  "machineName": "test",
  "name": "test",
  "submissionAccess": [],
  "tags": [],
  "title": "Test",
  "type": "form",
}
`;
